src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -884,6 +884,16 @@ public static decimal Max(decimal val1, decimal val2)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static double Max(double val1, double val2)
         {
+            if (AdvSimd.Arm64.IsSupported &&
+                (!RuntimeHelpers.IsKnownConstant(val1) || !RuntimeHelpers.IsKnownConstant(val1)))
+            {
+                // Unlike x86, ARM has a hw instruction for IEEE 754:2019 `maximum`
+                // If both val1 and val2 are constants - let the JIT fold it without this path
+                return AdvSimd.Arm64.MaxNumber(
+                    Vector128.CreateScalarUnsafe(val1),
+                    Vector128.CreateScalarUnsafe(val2)).ToScalar();
+            }
+
```

Comment:
We just want `AdvSimd.Arm64.MinScalar` and we want it in all scenarios (constant or not) since its a single instruction.

Since its an optimization to a single instruction, we should likely also handle it as an intrinsic on ARM64. It will help with throughput (less trees, no need to deal with inlining, etc) and likely overall perf and codegen.

Reply:
For reference, `Max` propagates back `NaN` if one input is `NaN`. `MaxNumber` on the other hand propagates back a `number` if one input is `NaN`.

Both are IEEE 754 defined functions